libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Input,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { KeyboardSupportItemInterface } from '../../utils/interfaces/keyboard-support-item.interface';
+import { ButtonComponent } from '../../button/button.component';
+
+
+
+/**
+ * A component used to enforce a certain layout for the action sheet.
+ * ```html
+ * <fd-action-sheet>
+ *     <fd-action-sheet-control>Control Element</fd-action-sheet-control>
+ *     <fd-action-sheet-body>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *     </fd-action-sheet-body>
+ * </fd-action-sheet>
+ * ```
+ */
+@Component({
+    selector: 'fd-action-sheet-item',
+    templateUrl: './action-sheet-item.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        'class': 'fd-action-sheet__item'
+    }
+})
+
+export class ActionSheetItemComponent implements KeyboardSupportItemInterface {
+
+    /** Sets text of button. */
+    @Input()
+    label: string;
+
+    /** Sets icon of action item. */
+    @Input()
+    glyph: string;
+
+    /** Indicate state of the button.*/
+    @Input()
+    negative = false;
+
+    /** Indicate if items should be in compact or compare mode. **/
+    @Input()
+    compact = false;
+
+    /** Display in mobile view. **/
+    @Input()
+    mobile = false;
+
+    /** Display in mobile view. **/
+    @Input()
+    isCloseButton = false;
+
+    /** Display in mobile view. **/
+    @Input()
+    close: any;
+
+    /** @hidden */
+    @ViewChild(ButtonComponent)
+    buttonComponent: ButtonComponent;
+
+    /** @hidden Implementation of KeyboardSupportItemInterface | TODO Revisit KeyboardSupportItemInterface*/
+    @Output()
+    keyDown: EventEmitter<KeyboardEvent> = new EventEmitter<KeyboardEvent>();
+
+    /** Event emitted, when the combobox's popover body is opened or closed */
+    @Output()
+    readonly openChange: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
I doubt action sheet item needs to know about open state, end user can subscribe to native `(click)` event from this element, to trigger open state by himself